<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="0.0"/>
  <arg name="y_pos" default="0.0"/>
  <arg name="z_pos" default="0.0"/>
  <arg name="roll" default="0"/>
  <arg name="pitch" default="0"/>
  <arg name="yaw" default="1.54"/>
  <arg name="init_state" default="0 0 0"/>
  <arg name="goal_state" default="4.8 -0.5 0"/>
  <arg name="rpm1" default="10"/>
  <arg name="rpm2" default="15"/>
  <arg name="obj_clear" default="5"/>
  <arg name="robot_clear" default="20"/>
  <arg name="node_start_delay" default="10.0" />  
  
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find astar_turtlebot)/world/map.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model turtlebot3_$(arg model) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -R $(arg roll) -P $(arg pitch) -Y $(arg yaw) -param robot_description" />

  <node pkg="astar_turtlebot" type="turtlebot_controller.py" name="astar_publisher" output="screen" />
  <node pkg="astar_turtlebot" type="turtlebot_pathgen.py" name="astar_planner" output="screen" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
    <!-- <param name="InitState" value="$(arg init_state)"/>
    <param name="GoalState" value="$(arg goal_state)"/>
    <param name="LeftRPM" value="$(arg rpm1)"/>
    <param name="RightRPM" value="$(arg rpm2)"/>
    <param name="ObjectClearance" value="$(arg obj_clear)"/>
    <param name="RobotClearance" value="$(arg robot_clear)"/> -->
  </node>

  
  
</launch>

